engine/link/link.asm

```diff
@@ -1882,23 +1882,29 @@ LinkTrade:
 	call LoadTradeScreenBorder
 	call SetTradeRoomBGPals
 	farcall Link_WaitBGMap
-	ld b, $1
+
+; Check if either of the pokemon sent was a Mew or Celebi, and send a different
+; byte depending on that. Presumably this would've been some prevention against
+; illicit trade machines, but it doesn't seem like a very effective one.
```

Comment:
"Presumably this could've been some prevention against illicit trade machines, but it doesn't actually do anything." (It doesn't do anything, right? You can trade Mew or Celebi like any other Pokémon.)

Reply:
You can trade these pokemon between devices that send the right byte here, which all gen2 games do, and it's ignored when linking with gen1 games.
My best guess is that the intention was to trip up people that made software or hardware to trade with gen2 games, that wouldn't have sent the right byte here if they didn't specifically test this scenario. To that effect it certainly does _something_. It's just not necessary to perform a trade, as this specific trade will still work between two games that remove this code.

Reply:
Okay, then how about another comment on the `cp b` below explaining that the two games' sent bytes have to match?

Reply:
I've clarified this in a comment right before the loop. Other than that this should be obvious from the context (set byte to send, exchange, check received byte).